irc_client/gui.py

```diff
@@ -269,10 +269,15 @@ def remove_view(self, view: ChannelView | PMView) -> None:
                 or self.view_selector.prev(view.view_id)
             )
         self.view_selector.delete(view.view_id)
+        view.stop_logging()
         view.destroy_widgets()
         del self.views_by_id[view.view_id]
 
     def remove_server(self, server_view: ServerView) -> None:
+        for subview in server_view.get_subviews(include_server=True):
+            subview.stop_logging()
+            subview.destroy_widgets()
+
         del self.views_by_id[server_view.view_id]
```

Comment:
- Subviews are not deleted from `self.views_by_id`
- No "select something else" logic

Reply:
Making separate PR